zerver/tests/test_auth_backends.py

```diff
@@ -1695,6 +1695,221 @@ def test_github_oauth2_email_not_associated(self) -> None:
+        for stream_name in set(stream_names):
+            stream = ensure_stream(realm, stream_name)
+            streams.append(stream)
+
+        # Without the invite link, we can't create an account due to invite_required
+        result = self.get_log_into_subdomain(data)
+        self.assertEqual(result.status_code, 200)
+        self.assert_in_success_response(['Sign up for Zulip'], result)
+
+        # Now confirm an invitation link works
+        referrer = self.example_user("hamlet")
+        multiuse_obj = MultiuseInvite.objects.create(realm=realm, referred_by=referrer)
+        multiuse_obj.streams.set(streams)
+        create_confirmation_link(multiuse_obj, realm.host, Confirmation.MULTIUSE_INVITE)
+        multiuse_confirmation = Confirmation.objects.all().last()
+        multiuse_object_key = multiuse_confirmation.confirmation_key
+
+        data["multiuse_object_key"] = multiuse_object_key
+        result = self.get_log_into_subdomain(data)
+        self.assertEqual(result.status_code, 302)
+
+        confirmation = Confirmation.objects.all().last()
+        confirmation_key = confirmation.confirmation_key
+        self.assertIn('do_confirm/' + confirmation_key, result.url)
+        result = self.client_get(result.url)
+        self.assert_in_response('action="/accounts/register/"', result)
+        data2 = {"from_confirmation": "1",
+                 "full_name": data['name'],
+                 "key": confirmation_key}
+        result = self.client_post('/accounts/register/', data2, subdomain="zulip")
+        self.assert_in_response("We just need you to do one last thing", result)
+
+        # Verify that the user is asked for name but not password
+        self.assert_not_in_success_response(['id_password'], result)
+        self.assert_in_success_response(['id_full_name'], result)
+
+        # Click confirm registration button.
+        result = self.client_post(
+            '/accounts/register/',
+            {'full_name': 'New User Name',
+             'key': confirmation_key,
+             'terms': True})
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(sorted(self.get_streams('new@zulip.com', realm)), stream_names)
+
+    def test_log_into_subdomain_when_email_is_none(self) -> None:
+        data = {'name': None,
+                'email': None,
+                'subdomain': 'zulip',
+                'is_signup': False,
+                'next': ''}
+
+        with mock.patch('logging.warning'):
+            result = self.get_log_into_subdomain(data)
+            self.assert_in_success_response(["You need an invitation to join this organization."], result)
+
+    def test_user_cannot_log_into_wrong_subdomain_with_cookie(self) -> None:
+        data = {'name': 'Full Name',
+                'email': self.example_email("hamlet"),
+                'subdomain': 'zephyr'}
+        result = self.get_log_into_subdomain(data)
+        self.assert_json_error(result, "Invalid subdomain")
+
```

Comment:
Do we need all these tests? Aren't the ones that will be run due to us subclassing ``SocialAuthBase`` sufficient?

Reply:
I'm not sure, I got everything from `GoogleAuthBackendTest`. The ones run from `SocialAuthBase` might be sufficient. Do we need anything else other than the ones run from `SocialAuthBase`.

Reply:
Unless there's some gitlab-specific stuff that needs testing, I think the general tests from SocialAuthBase should be sufficient. Although now looking at these "log into subdomain" tests I remember noticing before that they're only in the Google tests, while I think that's a mistake and they should be with other general tests. Can you try to move them to SocialAuthBase in a follow-up commit?
And either way, I think it's best to remove them from the Gitlab tests.

Reply:
removed the GitLab tests. Will try moving the google tests to SocialAuthBase this weekend.